pkgs/development/libraries/cogl/default.nix

```diff
@@ -35,26 +55,47 @@ stdenv.mkDerivation rec {
 
   configureFlags = [
     "--enable-introspection"
+  ] ++ lib.optionals (!stdenv.isDarwin) [
     "--enable-kms-egl-platform"
     "--enable-wayland-egl-platform"
     "--enable-wayland-egl-server"
-  ] ++ lib.optional gstreamerSupport "--enable-cogl-gst"
-  ++ lib.optionals (!stdenv.isDarwin) [ "--enable-gles1" "--enable-gles2" ];
+  ] ++ lib.optionals gstreamerSupport [
+    "--enable-cogl-gst"
+  ] ++ lib.optionals (!stdenv.isDarwin) [
+    "--enable-gles1"
+    "--enable-gles2"
+  ] ++ lib.optionals stdenv.isDarwin [
+    "--disable-glx"
+    "--without-x"
+  ];
 
   # TODO: this shouldn't propagate so many things
   # especially not gobject-introspection
-  propagatedBuildInputs = with xorg; [
-      glib gdk-pixbuf gobject-introspection wayland mesa
-      libGL libXrandr libXfixes libXcomposite libXdamage
-    ]
-    ++ lib.optionals gstreamerSupport [ gst_all_1.gstreamer
-                                               gst_all_1.gst-plugins-base ];
+  propagatedBuildInputs = [
```

Comment:
Do we really must propagate all of this?

Reply:
I think the only one I can remove without massive changes is `gobject-introspection`. Mainly,

- the `lib.optionals stdenv.isLinux` should be propagated by `cogl`, and,
- the `gst_all_1` ones should _at least_ be propagated by `clutter-gtk`.

For the others, i.e. `glib` and `gdk-pixbuf`, they should either be propagated by `clutter` or be included in all the packages having `clutter` as dependency, and this includes the whole Cinnamon desktop...

Reply:
Anyway, I would prefer that the Linux guys figure them out.

Reply:
Let’s just keep it as is, unless someone wants to figure that out. cogl and clutter are dead projects, anyway.